### PR TITLE
build: remove useless PHONY and deps

### PR DESCRIPTION
**Pull request by Rustin-Liu**
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

### What problem does this PR solve?

Problem Summary:

None

### What is changed and how it works?

What's Changed:
Remove useless PHONY and deps.

How it Works:
Remove useless PHONY and deps.
### Related changes

None

### Check List 

Tests 

- Unit test
- Integration test

Side effects

None

### Release note 
- Remove useless PHONY and deps

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@zz-jason @tiancaiamao  PTAL~

File: Makefile [hunk 1]
```diff
@@ -295,12 +295,6 @@ tools/bin/unconvert: tools/check/go.mod
 tools/bin/failpoint-ctl: go.mod
 	$(GO) build -o $@ github.com/pingcap/failpoint/failpoint-ctl
 
-tools/bin/misspell:tools/check/go.mod
```
Comment (tiancaiamao) [inline Makefile:298]: Why is it useless?
https://github.com/pingcap/tidb/pull/16834/files#diff-b67911656ef5d18c4ae36cb6741b7965L103

Comment (Rustin-Liu) [reply Makefile:298]: Because golangci-lint already already has these plugins internally.(such as misspell, ineffassign...)

**Comment by Rustin-Liu** [comment]
@tiancaiamao  @zz-jason  PING~

**Comment by Rustin-Liu** [comment]
Hi @zz-jason  @tiancaiamao, any possible to merge this?

**Comment by zz-jason** [comment]
> Hi @zz-jason @tiancaiamao, any possible to merge this?

Sorry for the late reply, LGTM now.

**Comment by Rustin-Liu** [comment]
@tiancaiamao  Ping~

**Comment by Rustin-Liu** [comment]
@sre-bot  /rebuild

**Comment by Rustin-Liu** [comment]
@sre-bot /rebuild

**Comment by Rustin-Liu** [comment]
@sre-bot /merge

**Comment by zz-jason** [comment]
Seems CI is failed, we are investigating the root cause.

**Pull request merged**